src/Core/src/Platform/iOS/WindowViewController.cs

```diff
@@ -0,0 +1,144 @@
+#if MACCATALYST
+using System;
+using System.Diagnostics.CodeAnalysis;
+using CoreGraphics;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal class WindowViewController : UIViewController
+{
+	WeakReference<IView?> _iTitleBarRef;
+	bool _isTitleBarVisible = true;
+
+	[UnconditionalSuppressMessage("Memory", "MEM0002", Justification = "Proven safe in device test: 'TitleBar Does Not Leak'")]
+	UIView? _titleBar;
+
+	[UnconditionalSuppressMessage("Memory", "MEM0002", Justification = "Proven safe in device test: 'TitleBar Does Not Leak'")]
+	UIView? _contentWrapperView;
+
+	/// <summary>
+	/// Instantiate a new <see cref="WindowViewController"/> object.
+	/// </summary>
+	/// <param name="windowContentViewController">An instance of the <see cref="UIViewController"/> that is the RootViewController.</param>
+	/// <param name="window">An instance of the <see cref="IWindow"/>.</param>
+	/// <param name="mauiContext">An instance of the <see cref="IMauiContext"/>.</param>
+	/// <remarks>
+	/// Only dragging the top of the titlebar will move the window.
+	/// The top of the TitleBar will also drag the window inside of elements like buttons.
+	/// Gestures such as swiping and controls like swipeview will not work inside the TitleBar.
+	/// </remarks>
+	public WindowViewController(UIViewController windowContentViewController, IWindow window, IMauiContext mauiContext)
+	{
+		_iTitleBarRef = new WeakReference<IView?>(null);
+
+		// Note: Maintain the order for adding a new ViewController to a Container ViewController
+		// 1. Add the Subview
+		// 2. Arrange the Subview's frame
+		// 3. AddChildViewController
+		// 4. Call DidMoveToParentViewController
+
+		if (View is not null && windowContentViewController.View is not null)
+		{
+			_contentWrapperView = new();
+			View.AddSubview(_contentWrapperView);
+			_contentWrapperView.AddSubview(windowContentViewController.View);
+		}
+
+		SetUpTitleBar(window, mauiContext, true);
+		AddChildViewController(windowContentViewController);
+		windowContentViewController.DidMoveToParentViewController(this);
```

Comment:
Do we need this? Calling life cycle events manually feels scary.

Reply:
Yeah this was related to that discussion we had about the NavigatedTo and other lifecycle events not propagating to the new view that is inside the _contentWrapperView and it sounded like this is the standard for creating new ContainerViewControllers specifically from the "Adding a Child View Controller to Your Content" section [here](https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/ImplementingaContainerViewController.html).